src/Stripe.Tests.XUnit/coupons_new/coupons.cs

```diff
@@ -0,0 +1,116 @@
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Xunit;
+
+namespace Stripe.Tests.Xunit
+{
+    public class coupon_test
+    {
+        private string previousApiBase;
+        private readonly string resourceId = "co_123";
+        private StripeCouponService service;
+
+        public coupon_test()
+        {
+            this.service = new StripeCouponService(Cache.ApiKey);
+            this.previousApiBase = StripeConfiguration.GetApiBase();
+            StripeConfiguration.SetApiBase("http://localhost:12111/v1");
```

Comment:
Just as a quick sanity check: is there any possibility that tests run in parallel? If so, this isn't safe.

Reply:
As far as I could tell it's not happening no, and I confirmed that tests fail if I don't clean up after the destruction, so I think it works! We will definitely confirm 100% before merging anything

Reply:
[Your choice](https://stackoverflow.com/questions/1408175/execute-unit-tests-serially-rather-than-in-parallel). If possible, shoot for 100% isolation imo. 👍